src/main/java/fi/helsinki/cs/tmc/core/communication/oauth2/PasswordFlow.java

```diff
@@ -0,0 +1,41 @@
+package fi.helsinki.cs.tmc.core.communication.oauth2;
+
+import fi.helsinki.cs.tmc.core.configuration.TmcSettings;
+import fi.helsinki.cs.tmc.core.holders.TmcSettingsHolder;
+
+import org.apache.oltu.oauth2.client.OAuthClient;
+import org.apache.oltu.oauth2.client.URLConnectionClient;
+import org.apache.oltu.oauth2.client.request.OAuthClientRequest;
+import org.apache.oltu.oauth2.client.response.OAuthJSONAccessTokenResponse;
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+import org.apache.oltu.oauth2.common.message.types.GrantType;
+
+/**
+ * A password flow for oauth.
+ *
+ * <p>Password flow fetches the token from server by using username and password</p>
+ */
+public class PasswordFlow implements OauthFlow {
+
+    public PasswordFlow() {
+    }
+
+    @Override
+    public String getToken() throws OAuthSystemException, OAuthProblemException {
+        TmcSettings settings = TmcSettingsHolder.get();
+        OAuthClientRequest request = OAuthClientRequest
+                .tokenLocation(settings.getOauthTokenUrl())
+                .setGrantType(GrantType.PASSWORD)
+                .setClientId(settings.getOauthApplicationId())
+                .setClientSecret(settings.getOauthSecret())
+                .setUsername(settings.getUsername())
+                .setPassword(settings.getPassword())
+                .buildQueryMessage();
+        OAuthClient client = new OAuthClient(new URLConnectionClient());
+        String token = client.accessToken(request, OAuthJSONAccessTokenResponse.class)
```

Comment:
Can you be more specific about why and what should be logged?
I can only assume token needs to be outputted for some debugging purposes.
Also an earlier comment was asking how to log.